tests/e2e/test_tracking.py

```
@@ -148,6 +148,21 @@ def test_multiple_objects_tracked(self, model, video_path):
 
+    def test_ocsort_tracker_end_to_end(self, model, video_path):
+        """The OC-SORT tracker also yields stable, unique per-frame IDs."""
+        frames = _run_tracker(model, video_path, n_frames=20, tracker="ocsort")
+        assert len(frames) == 20
+        stable = 0
+        for i, f in enumerate(frames):
+            ids = f.track_id.tolist() if f.track_id is not None else []
+            assert len(ids) == len(set(ids)), f"Frame {i}: duplicate IDs: {ids}"
+            if i > 0:
+                prev = _ids(frames[i - 1])
+                curr = _ids(frames[i])
```

Ruff (0.15.15)

[warning] List index lookup in `enumerate()` loop

Use the loop variable directly

(PLR1736)